src/common/libflux/disconnect.h

```diff
@@ -17,11 +17,23 @@
 extern "C" {
 #endif
 
+/* Convenience function for ensuring messages match for disconnect.
+ * Effectively calls flux_msg_match_route_first() and
+ * flux_msg_cred_authorize().
+ */
+bool flux_disconnect_match (const flux_msg_t *msg1, const flux_msg_t *msg2);
+
 /* Remove all messages in 'l' with the same sender as 'msg'.
  * Return 0 or the number of messages removed.
  */
 int flux_msglist_disconnect (struct flux_msglist *l, const flux_msg_t *msg);
 
+/* Convenience function for ensuring messages match for cancel.
+ * Effectively calls flux_disconnect_match() and flux_msg_get_matchtag() to
+ * compare matchtags.
+ */
```

Comment:
How about "Return true if cancel request `msg1` came from same sender as `msg2`, has appropriate authorization, and references the matchtag in `msg2`.